docusaurus-docs/conda-store/references/deprecation-guide.md

```diff
@@ -0,0 +1,36 @@
+---
+description: Deprecation guide
+---
+
+# Deprecation guide
+
+This document outlines how conda-store deprecates functionality. Deprecation notices will always be available in release notes.
+
+## Deprecations in the REST API
+
+Deprecations in the REST API follow the form outlined by the [deprecation header RFC](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-deprecation-header-02).
+
+### Deprecating routes/endpoints
+
+An endpoint that is deprecated will include the response headers:
+```
+{
+   "Deprecation": "True",
+   "Sunset": <removal date, eg. "Mon, 16 Feb 2025 23:59:59 UTC" >
+}
+```
+
+The "removal date" indicates the date after which conda store will no longer serve the endpoint. It will be specified as a [HTTP-Date](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date)
```

Comment:
Mmmm, we already have a ["Removing versions of API endpoints"] policy (https://deploy-preview-1018--conda-store.netlify.app/community/policies/backwards-compatibility#removing-versions-of-api-endpoints).

It might be worth merging these two:
- I like the addition of the `Deprecation` header, but in the current policy, we have `removal_reason` that would contain natural deprecation cycles and other reasons for removal. @dcmcand would it be too redundant to have both?
- I also prefer the newly proposed date format (`HTTP-date`) vs. `YYYY-MM-DD`

Reply:
Hmm, I think the deprecation step and removal step should be separate. Like all the steps to removing an API endpoint would be: 

1. mark the endpoint as deprecated. All functionality should stay the same, but the response headers should include
```
{
   "Deprecation": "True",
   "Sunset": <removal date, eg. "Mon, 16 Feb 2025 23:59:59 UTC" >
}
```
2. after 1 release, update the endpoint to return `410 Gone` with return payload specified in the linked doc
3. after 1 more release fully remove the endpoint